Processes.md

```diff
@@ -21,42 +21,114 @@ Up to date
 Maintain state.
 
 
-Let’s create a module Menuprocess.ex and see how we shall execute the process from the client in the menuserver.
+Let’s create a module MenuProcess.ex and see how we shall execute the process from the client in the menuserver.
 Below is a breakdown of a couple of processes in our module. Our module will enable three processes;
 Return the current menu state
 Add items to the menu
 Handle wrong entries
 
-For the client requests to be handled on the server, start_link is called, the function menu_state is spawned to its own
-process then passed the initial state which in our case will change from an empty [] to an actual list of items to be 
-added to the menu. The function responsible for that is shown below;
+start_link begins a spawn linked to the current process with the given functions, the pid is registered using `Process.register`. For the client requests to be handled on the server, start_link is called, the function menu_state is spawned to its own process then passed the initial state which in our case will change from an empty [] to an actual list of items to be added to the menu. The function responsible for that is shown below;
 
-**add code**
+ ```
+defmodule Menuprocess do
+  
+  #menu server
+  @doc"""
+  This function initializes our server 
+  """
+   @spec start_link(any()) :: pid()
+    def start_link(initial_state \\ []) do
+     pid = spawn(__MODULE__, :menu_state, [initial_state])
+     Process.register(pid, :process)
+     pid 
+    end
+end 
+```
 
 Return the current menu state:
```

Comment:
What returns the current menu context, I would try and provide more context to keep the reader involved